File: 3242-count-elements-with-maximum-frequency/3242-count-elements-with-maximum-frequency.py
class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        frequency = [0] * (max(nums) + 1)
        for i in range(len(nums)):
            frequency[nums[i]] += 1
        heap = []
        for i in range(len(frequency)):
            heapq.heappush(heap,(-frequency[i],i))
        max_element = heap[0][0]
        cnt = 0
        while len(heap) != 0:
            if max_element == heap[0][0]:
                cnt += (-heap[0][0])
                heapq.heappop(heap)
            else:
                return cnt
        return -1

File: 3242-count-elements-with-maximum-frequency/3242-count-elements-with-maximum-frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        int answer = 0;
        int maximumValue = -1;
        int getMaximum = -1;
        int size = nums.length;

        for(int i = 0;i < size;i++){
            if(nums[i] > getMaximum){
                getMaximum = nums[i];
            }
        }

        int [] frequency = new int[getMaximum + 1];

        for(int i = 0; i < size; i++){
            frequency[nums[i]]++;
        }

        for(int i = 0; i < frequency.length;i++){
            if(frequency[i] > maximumValue){
                maximumValue = frequency[i];
            }
        }

        for(int i = 0; i < frequency.length; i++){
            if(frequency[i] == maximumValue){
                answer += maximumValue;
            }
        }
        return answer;
    }
}